### PR TITLE
feat(snippet-bot): read products from service configs in googleapis/googleapis

### LINKED ISSUE #4994: bug: snippet bot flagging "kmsinventory" as an invalid product prefix
I imagine this is because it thinks that `kmsinventory` is a subset of `kms`? Are other languages combining this into one package?

http://cloud.google.com/kms/docs/reference/inventory/rest
https://github.com/googleapis/google-cloud-php/pull/5942

### PR DESCRIPTION
**Pull request by chingor13**
Short term solution to merge new products from service_configs in googleapis/googleapis into the exported product list from DKP.

To improve performance, we use a daily cron job to scan googleapis/googleapis for service configs and merge the result with the exported list from DKP.

Fixes #4994

feat: add daily cron to reload products cache
refactor: pass ApiLabels list into violations logic to avoid implicit API call

### CODE REVIEW

File: packages/snippet-bot/src/violations.ts [hunk 1]
```diff
@@ -198,10 +198,9 @@ const isAutoGenerated = (regionTag: string): boolean => {
 export const checkProductPrefixViolations = async (
   changes: ChangesInPullRequest,
   config: Configuration,
-  logger: GCFLogger = defaultLogger
+  apiLabels: ApiLabels
 ): Promise<Array<Violation>> => {
   const ret: Violation[] = [];
-  const apiLabels = await getApiLabels(dataBucket, logger);
```
Comment (chingor13) [inline packages/snippet-bot/src/violations.ts:204]: Avoid an implicit API call here -- explicitly provide the list of allowed products. The caller is expected to source the list.

File: packages/snippet-bot/src/api-labels.ts [hunk 2]
```diff
@@ -28,15 +31,51 @@ export interface ApiLabels {
   products: Array<ApiLabel>;
 }
 
+export const getDriftApiLabels = async (
```
Comment (SurferJeffAtGoogle) [inline packages/snippet-bot/src/api-labels.ts:34]: Please add a comment for this getDriftApiLabels and mergeApiLabels.

File: packages/snippet-bot/src/service-configs.ts [hunk 3]
```diff
@@ -0,0 +1,82 @@
...
+// limitations under the License.
+
+/* eslint-disable-next-line node/no-extraneous-import */
+import {Octokit} from '@octokit/rest';
+import {RepositoryFileCache} from '@google-automations/git-file-utils';
+import yaml from 'js-yaml';
+import {ApiLabels, ApiLabel} from './api-labels';
+import {GCFLogger, logger as defaultLogger} from 'gcf-utils';
+
+interface ServiceConfig {
```
Comment (SurferJeffAtGoogle) [inline packages/snippet-bot/src/service-configs.ts:22]: What kind of service?  Comment please.

File: packages/snippet-bot/src/service-configs.ts [hunk 4]
```diff
@@ -0,0 +1,82 @@
...
+    github_label: string;
+    doc_tag_prefix: string;
+  };
+}
+
+interface ScanLabelsOptions {
+  branch?: string;
+  logger?: GCFLogger;
+}
+export const scanServiceConfigsForApiLabels = async (
```
Comment (SurferJeffAtGoogle) [inline packages/snippet-bot/src/service-configs.ts:39]: Comments please.

**Review by SurferJeffAtGoogle — CHANGES_REQUESTED** [review]

File: packages/snippet-bot/src/api-labels.ts (hunk 2 above)
Comment (chingor13) [reply packages/snippet-bot/src/api-labels.ts]: Done.

File: packages/snippet-bot/src/service-configs.ts (hunk 4 above)
Comment (chingor13) [reply packages/snippet-bot/src/service-configs.ts]: Done.

**Review by SurferJeffAtGoogle — APPROVED** [review]

**Pull request merged**